test/language/expressions/greater-than-or-equal/bigint-and-number-extremes.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2017 Josh Wolfe. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Comparisons of BigInt and Number values
+esid: sec-abstract-relational-comparison
+info: |
+  ...
+  3. If both px and py are Strings, then
+    ...
+  4. Else,
+    a. Let nx be ? ToNumeric(px). Because px and py are primitive values evaluation order is not important.
+    b. Let ny be ? ToNumeric(py).
+    c. If Type(nx) is Type(ny), return ? Type(nx)::lessThan(nx, ny).
+    d. Assert: Type(nx) is BigInt and Type(ny) is Number, or if Type(nx) is Number and Type(ny) is BigInt.
+    e. If x or y are any of NaN, return undefined.
+    f. If x is -∞, or y is +∞, return true.
+    g. If x is +∞, or y is -∞, return false.
+    h. If the mathematical value of nx is less than the mathematical value of ny, return true, otherwise return false.
+features: [BigInt]
+---*/
+
+assert.sameValue(1n >= Number.MAX_VALUE, false);
+assert.sameValue(Number.MAX_VALUE >= 1n, true);
+assert.sameValue(1n >= -Number.MAX_VALUE, true);
+assert.sameValue(-Number.MAX_VALUE >= 1n, false);
+assert.sameValue(
+  0xfffffffffffff80000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000n >= Number.MAX_VALUE,
```

Comment:
Would it make sense to have a test for the edge cases, namely the BigInt which represents one less than Number.MAX_VALUE, the BigInt which is Number.MAX_VALUE, and one greater than it? Also, for mixed Number/BigInt comparison tests, it would be great to have more tests of different magnitudes to get at what are likely to be different cases in implementations (e.g., test on a Number which is between 0 and 1, and therefore can't be represented faithfully as a BigInt).

Reply:
I addressed this in the OP of this PR, but it might be good to throw a comment in the file as well. What do you think?

Reply:
Sorry, I missed the explanation there. I disagree with your reading of the spec--I think Number.MAX_VALUE is well-defined.

Reply:
Alright, I'll tighten up the test.